src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -3,58 +3,53 @@ package org.broadinstitute.dsde.workbench.sam.service
 
-  def acceptTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[Boolean]] =
-    if (tosConfig.enabled) {
-      directoryDao.acceptTermsOfService(userId, tosConfig.version, samRequestContext).map(Option(_))
-    } else IO.pure(None)
+  def acceptTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] =
+    directoryDao
+      .acceptTermsOfService(userId, tosConfig.version, samRequestContext)
+      .withInfoLogMessage(s"$userId has accepted version ${tosConfig.version} of the Terms of Service")
 
-  def rejectTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[Boolean]] =
-    if (tosConfig.enabled) {
-      directoryDao.rejectTermsOfService(userId, samRequestContext).map(Option(_))
-    } else IO.pure(None)
+  def rejectTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] =
+    directoryDao
+      .rejectTermsOfService(userId, samRequestContext)
+      .withInfoLogMessage(s"$userId has rejected version ${tosConfig.version} of the Terms of Service")
 
-  def getTosDetails(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[TermsOfServiceDetails] =
-    for {
-      userAcceptedVersion <- getUserAcceptedVersion(userId, samRequestContext)
-    } yield TermsOfServiceDetails(tosConfig.enabled, tosConfig.isGracePeriodEnabled, tosConfig.version, userAcceptedVersion.flatten)
+  // This method will disappear once UI is using getTosAdherenceDetails
+  def getTosDetails(samUser: SamUser): IO[TermsOfServiceDetails] =
+    IO.pure(TermsOfServiceDetails(isEnabled = true, tosConfig.isGracePeriodEnabled, tosConfig.version, samUser.acceptedTosVersion))
 
-  /** Check if Terms of service is enabled and if the user has accepted the latest version
-    * @return
-    *   IO[Some(true)] if ToS is enabled and the user has accepted IO[Some(false)] if ToS is enabled and the user hasn't accepted IO[None] if ToS is disabled
-    */
-  def getTosStatus(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[Boolean]] =
-    if (tosConfig.enabled) {
-      getUserAcceptedVersion(userId, samRequestContext).map(_.map(_.contains(tosConfig.version)))
-    } else IO.none
-
-  def getUserAcceptedVersion(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[Option[String]]] =
-    directoryDao.loadUser(userId, samRequestContext).map(_.map(_.acceptedTosVersion))
+  def getTosAdherenceStatus(samUser: SamUser): IO[TermsOfServiceAdherenceStatus] = {
+    val userHasAcceptedLatestVersion = userHasAcceptedLatestTosVersion(samUser)
+    val canUseTerra = tosAcceptanceAllowsTerraUsage(samUser)
+    IO.pure(TermsOfServiceAdherenceStatus(samUser.id, userHasAcceptedLatestVersion, canUseTerra))
+  }
 
   /** If grace period enabled, don't check ToS, return true If ToS disabled, return true Otherwise return true if user has accepted ToS, or is a service account
```

Comment:
```suggestion
  /** IF user has accepted Current ToS, THEN return TRUE 
     * ELSE IF grace period enabled AND user has accepted ANY ToS, THEN return TRUE
     * ELSE IF user is a service account, THEN return TRUE
     * ELSE return FALSE
     */
```
I wonder if it might be easier to just code it like this comment is written out, then I think you can get ride of all of the inline comments making the logic easier to understand (in my opinion) and making the code self documenting.

Reply:
returning `true/false` in `if/else` is a code smell and will get flagged by SonarCloud.

Reply:
re-written to be clearer

Reply:
NICE!  I like this a lot!